src/coroutine/memoset/env.rs

```diff
@@ -94,21 +95,25 @@ impl<F: LurkField> Query<F> for EnvQuery<F> {
         }
     }
 
-    fn dummy_from_index(s: &Store<F>, index: usize) -> Self {
+    fn dummy_from_index(_: &Self::C, s: &Store<F>, index: usize) -> Self {
         match index {
             0 => Self::Lookup(s.num(0.into()), s.num(0.into())),
             _ => unreachable!(),
         }
     }
 
-    fn index(&self) -> usize {
+    fn query_rc(_: &Self::C, _index: usize) -> usize {
+        1
```

Comment:
What's the logic behind this choice? I'm not sure we are going to want this to be a decision made in the query implementation. The question here somewhat highlights why that might be.